parity/cli/subcommands.rs

```diff
@@ -0,0 +1,242 @@
+    )]
+    pub from: String,
+
+    #[structopt(
+        long,
+        name = "TO_BLOCK",
+        help = "Export to (including TO_BLOCK) block TO_BLOCK, which may be an index, hash or 'latest'",
+        default_value = "latest"
+    )]
+    pub to: String,
+    #[structopt(help = "Path to the exported file", name = "FILE")]
+    pub file: Option<String>,
+}
+
+#[derive(StructOpt, Deserialize, Debug, Clone)]
+#[structopt(
+    about = "Export the blockchain state from the given chain [default: mainnet] into a file. The command requires the chain to be synced with --fat-db on."
+)]
+pub struct ExportState {
+    #[structopt(long = "no-storage", help = "Don't export account storage.")]
+    pub no_storage: bool,
+
+    #[structopt(long = "no-code", help = "Don't export account code.")]
+    pub no_code: bool,
+
+    #[structopt(
+        long = "max-balance",
+        name = "MAX_WEI",
+        help = "Don't export accounts with balance greater than specified."
+    )]
+    pub max_balance: Option<String>,
+
+    #[structopt(
+        long = "min-balance",
+        name = "MIN_WEI",
+        help = "Don't export accounts with balance less than specified."
+    )]
+    pub min_balance: Option<String>,
+
+    #[structopt(
+        default_value = "latest",
+        long,
+        name = "BLOCK",
+        help = "Take a snapshot at the given block, which may be an index, hash, or latest. Note that taking snapshots at non-recent blocks will only work with --pruning archive"
+    )]
+    pub at: String,
+
+    #[structopt(
+        long,
+        name = "FORMAT",
+        help = "Export in a given format. FORMAT must be either 'hex' or 'binary'. [default: binary]"
+    )]
+    pub format: Option<String>,
+
+    #[structopt(long = "file", name = "FILE", help = "Path to the exported file")]
+    pub file: Option<String>,
+}
+
+#[derive(StructOpt, Deserialize, Debug, Clone)]
+#[structopt(about = "Manage Signer")]
+pub enum Signer {
+    #[structopt(
+        about = "Generate a new signer-authentication token for the given --chain (default: mainnet)"
+    )]
+    NewToken,
+    #[structopt(
+        about = "List the signer-authentication tokens from given --chain (default: mainnet)"
+    )]
+    List,
+    #[structopt(about = "Sign")]
+    Sign {
+        #[structopt(name = "ID")]
+        id: Option<usize>,
+    },
+    #[structopt(about = "Reject")]
+    Reject {
+        #[structopt(name = "ID")]
+        id: Option<usize>,
+    },
+}
+
+#[derive(StructOpt, Deserialize, Debug, Clone)]
+#[structopt(about = "Make a snapshot of the database of the given chain(default: mainnet)")]
+pub struct Snapshots {
+    #[structopt(
+        default_value = "latest",
+        name = "BLOCK",
+        help = "Take a snapshot at the given block, which may be an index, hash, or latest. Note that taking snapshots at non-recent blocks will only work with --pruning archive"
+    )]
+    pub at: String,
+    // FIXME: check if the default is correct
```

Comment:
How do you mean "default"? I don't think this should have a default (and it's required).